source/server/server.cc

```diff
@@ -147,6 +147,14 @@ int InstanceImpl::getListenSocketFd(const std::string& address) {
   return -1;
 }
 
+Network::ListenSocket* InstanceImpl::getListenSocketByIndex(uint32_t index) {
+  if (index < config_->listeners().size()) {
```

Comment:
nit: this is kind of an odd impl for this function (requiring advancing from beginning by N each time). I know this is only used in test code, but can we potentially implement with a for loop over list elements which also increments an index?